MekHQ/src/mekhq/campaign/autoresolve/acar/SimulationContext.java

```diff
@@ -0,0 +1,784 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ *  This file is part of MekHQ.
+ *
+ *  MekHQ is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  MekHQ is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.campaign.autoresolve.acar;
+
+
+import megamek.common.*;
+import megamek.common.actions.EntityAction;
+import megamek.common.annotations.Nullable;
+import megamek.common.enums.GamePhase;
+import megamek.common.enums.SkillLevel;
+import megamek.common.event.GameEvent;
+import megamek.common.event.GameListener;
+import megamek.common.force.Forces;
+import megamek.logging.MMLogger;
+import megamek.server.scriptedevent.TriggeredEvent;
+import megamek.server.victory.VictoryHelper;
+import megamek.server.victory.VictoryResult;
+import mekhq.campaign.autoresolve.acar.action.Action;
+import mekhq.campaign.autoresolve.acar.action.ActionHandler;
+import mekhq.campaign.autoresolve.acar.report.PublicReportEntry;
+import mekhq.campaign.autoresolve.component.AcTurn;
+import mekhq.campaign.autoresolve.component.Formation;
+import mekhq.campaign.autoresolve.component.FormationTurn;
+import mekhq.campaign.autoresolve.converter.SetupForces;
+import mekhq.campaign.autoresolve.event.AutoResolveConcludedEvent;
+import mekhq.campaign.mission.AtBScenario;
+import mekhq.campaign.mission.ScenarioObjective;
+import org.apache.commons.lang3.NotImplementedException;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * @author Luana Coppio
+ */
+public class SimulationContext implements IGame {
```

Comment:
This is a mm sourced class, so my input is going to be very limited.

@SJuliez I saw you reviewed the mm half of this. I don't suppose you'd mind taking a look at this specific class?

Reply:
I did multiple changes based on Juliez feedback and it improved it very much